### PR TITLE
fix(status-cards): delete posts without the streaming service [SOCIALPLAT-488]

### PR DESCRIPTION
**Pull request by jpezninjo**
## Goal

The streaming service stops working after a couple seconds. This causes posts to remain after the user deletes them (refreshing the page removes them) [SOCIALPLAT-488](https://mozilla-hub.atlassian.net/browse/SOCIALPLAT-488)

This PR (painfully) hooks into where posts are stored in our app and deletes them upon a successful API call.

## To Do:

- [x] Move the "delete status"-logic into it's own function out from the stream event handler
- [x] prop-drill function from `paginator.ts` to  `StatusActionMore.vue`
- [x] create a new `afterDelete` event; define it right next to the existing `afterEdit`
- [x] test it out

## Implementation Decisions
This doesn't do any special handling if the API fails and also is currently firing after the API finishes instead of before. This is on par with the [fix to Create Posts](https://github.com/MozillaSocial/elk/pull/42). The "before API"-behavior is how it should be to accomplish optimistic UI.

### CODE REVIEW

**Comment by bassrock** [comment]
@jpezninjo

> The streaming service stops working after a couple seconds
As in the Mastodon API closes the connection? Do we know if using elk.zone with another instance is this a problem?

**Comment by jpezninjo** [comment]
@bassrock good question! Just dived into this.

Streams get closed on our production site after a couple seconds
image




Same thing using mozilla.social on https://elk.zone/home

image




But using mastadon.social on elk.zone, everything is great. Note the timestamps showing activity beyond the estimated 5-8 seconds
image

File: components/status/StatusCard.vue [hunk 1]
```diff
@@ -171,7 +173,7 @@ const forceShow = ref(false)
                 </div>
               </div>
             </div>
-            <StatusActionsMore v-if="actions !== false" :status="status" me--2 />
+            <StatusActionsMore v-if="actions !== false" :status="status" me--2 @after-delete="props.deleteItem" />
```
Comment (jpezninjo) [inline components/status/StatusCard.vue:176]: I'm a fan now of using `emit`. I got convinced that this syntax makes it more clear that this child outputs a signal to its parent. Open to what others thing.

File: composables/paginator.ts [hunk 2]
```diff
@@ -35,6 +35,13 @@ export function usePaginator<T, P, U = T>(
     data.splice(0, 0, status)
   }
 
+  function deleteItem(id: string) {
+    const data = items.value as mastodon.v1.Status[]
```
Comment (jpezninjo) [inline composables/paginator.ts:39]: took these lines of code straight from the "delete" event handler below in this file

**Comment by bassrock** [comment]
@jpezninjo Given that it is broken on our instance, which is probably due to our HAProxy, can you instead raise this and the examples with the Platform team?

We should instead look at fixing the API here because all clients will have the same issue.

**Comment by jpezninjo** [comment]
@bassrock will do 👍 good call out about the other clients

**Pull request closed without merging**